Source/Tools/FEXServer/CoreDumpService.cpp

```diff
@@ -0,0 +1,337 @@
+          break;
+        }
+        case FEXServerClient::CoreDump::PacketTypes::DESC: {
+          FEXServerClient::CoreDump::PacketDescription *Req = reinterpret_cast<FEXServerClient::CoreDump::PacketDescription *>(&Data[CurrentOffset]);
+          SetDesc(Req->pid,
+            Req->tid,
+            Req->uid,
+            Req->gid,
+            Req->Signal,
+            Req->Timestamp,
+            Req->HostArch,
+            Req->GuestArch);
+
+          CurrentOffset += sizeof(*Req);
+          break;
+        }
+        case FEXServerClient::CoreDump::PacketTypes::HOST_CONTEXT: {
+          FEXServerClient::CoreDump::PacketHostContext *Req = reinterpret_cast<FEXServerClient::CoreDump::PacketHostContext *>(&Data[CurrentOffset]);
+          HostContextData = *Req;
+          // TODO: Create host context unwinder.
+          CurrentOffset += sizeof(*Req);
+          break;
+        }
+        case FEXServerClient::CoreDump::PacketTypes::GUEST_CONTEXT: {
+          FEXServerClient::CoreDump::PacketGuestContext *Req = reinterpret_cast<FEXServerClient::CoreDump::PacketGuestContext *>(&Data[CurrentOffset]);
+          Desc.GuestArch = Req->GuestArch;
+          // TODO: Create guest context unwinder.
+          CurrentOffset += sizeof(*Req);
+          break;
+        }
+        case FEXServerClient::CoreDump::PacketTypes::CONTEXT_UNWIND: {
+          // Print the header to the backtrace.
+          BacktraceHeader();
+          // TODO: Unwind the host and guest contexts that were received.
+          // Shutdown the client once FEXServer has unwound.
+          FEXServerClient::CoreDump::SendShutdownPacket(ServerSocket);
+
+          ShouldShutdown = true;
+          CurrentOffset += sizeof(FEXServerClient::CoreDump::PacketHeader);
+          break;
+        }
+        case FEXServerClient::CoreDump::PacketTypes::GET_JIT_REGIONS: {
+          FEXServerClient::CoreDump::PacketGetJITRegions *Req = reinterpret_cast<FEXServerClient::CoreDump::PacketGetJITRegions *>(&Data[CurrentOffset]);
+          ThreadDispatcher = Req->Dispatcher;
+          for (size_t i = 0; i < Req->NumJITRegions; ++i) {
+            ThreadJITRegions.emplace_back(Req->JITRegions[i]);
+          }
+          CurrentOffset += sizeof(*Req) + Req->NumJITRegions * sizeof(FEXCore::Context::Context::JITRegionPairs);
+          break;
+        }
+        default:
+          // Something sent us an invalid packet. To ensure we don't spin infinitely, consume all the data.
+          LogMan::Msg::EFmt("[CoreDumpClass] InvalidPacket size received 0x{:x} bytes", CurrentRead - CurrentOffset);
+          CurrentOffset = CurrentRead;
+          break;
+      }
+    }
+  }
+
+  void CoreDumpClass::ExecutionFunc() {
+    auto LastDataTime = std::chrono::system_clock::now();
+
+    while (!ShouldShutdown) {
+      struct timespec ts{};
+      ts.tv_sec = RequestTimeout;
+
+      int Result = ppoll(&PollFD, 1, &ts, nullptr);
```

Comment:
(Opening a new thread here since GitHub doesn't support threads for top-level PR comments...)

Any comment on my top-level remark https://github.com/FEX-Emu/FEX/pull/2432#issuecomment-1602900616 ?

I overlooked one further use of ppoll (in FEXServerClient.cpp), so this `while (!ShouldShutdown) { ... }` pattern is actually repeated four times, each time embedded in nontrivial application logic.

Reply:
Added a new SocketConnectionHandler and moved CoreDumpService and ProcessPipe over to it.
Didn't move Logger over to it since it is doing some more special stuff.
Can see if this works first.